### ISSUE TITLE
Gerald misalignment

### ISSUE THREAD

**Issue by DaveBenji** [DoESLiverpool/somebody-should#1904]
Gerald has a misalignment between the x and x axis. You cant cut a true circle. If you cut a complicated free-form, the beginning point and the end point are misaligned

**Comment by magman2112** [comment]
I did a few test cuts on Maker Night and I couldn’t find any issues. I tried a dimension test for both the x and y axis and cut a circle and the dimensions were accurate (to within 0.5mm).

could you possibly let me have an example of one of your files that you had issues with, so that I can see if I can replicate your issue?

@DoESLiverpool/laser-maintenance

**Comment by DaveBenji** [comment]
I'll take a few photos today. regards

**Comment by DaveBenji** [comment]
I had to abandon Gerald today as he's even more out of line. I was loosing work so I gave up.
I attach two photos of a cut made with the same drawing but at different XY locations and a you cab see on the second photo its really distorted and ruined.  I tried to attach the file but github doesnt support lightburn file type. I could email it? Regards
![sparrow01](https://github.com/DoESLiverpool/somebody-should/assets/130652231/bfbae0c9-8a0a-4e86-a5e4-fb33b91a4861)

![sparrow02](https://github.com/DoESLiverpool/somebody-should/assets/130652231/b93a288c-fb9c-465f-9c23-9ca38cc01349)

**Comment by DaveBenji** [comment]
![SPARROWS 300 X 100](https://github.com/DoESLiverpool/somebody-should/assets/130652231/bd107adb-87c4-4c21-8054-0183d50807ec)
INKSCAPE VERSION

**Comment by DaveBenji** [comment]
I plan to use Gerald on Wednesday. Has there been any update on this issue?

**Comment by magman2112** [comment]
Sorry Dave, I meant to comment on this yesterday.

@Sean-anotherone and I spent a lot of time on this issue on Thursday night.

Your file was a great help, as this allowed us to find a repeatable issue that we could try to diagnose.

The issue appears to be a minute creep on the X-axis (left to right. It appears to be on the fast movement between cuts, rather than on the cuts themselves, so was more apparent on the second layer (between the legs) than the main body cuts.

We knew that there was an open issue for the belt for the x-axis, so we fitted the new belt that has been sat in the laser spares cupboard for a few weeks. This made a slight difference, but didn’t eliminate the issue.

Next, we tested some changes to the step multiplier on the stepper motor drivers along with firmware changes to match. This gave us another incremental improvement, but it still isn’t perfect. Our thought is that some steps are getting lost by the older stepper drivers and by driving them with a different frequency from the controller board, we can make the losses less significant.

We are going to try replacing the x-axis stepper driver next, but that will be undertaken next Thursday night at the earliest.

I did find a workaround though, that might help. As Lightburn has lots of layers to play with, I split the two layers into columns. So the first column of cuts uses layers 1 and 2, the second column uses layer 3 and 4, etc. duplicating the settings from layer 1 to layer 3,5,7,etc. This keeps the cuts in a relatively short area to minimise the fast movement cumulative errors. This worked well with your test file (or a 4x6 matrix of your drawing we used as our test). This is a similar process we use with engraving in columns, to increase productivity when laser engraving (fill in Lightburn).

**Comment by DaveBenji** [comment]
Thanks for that. I'm glad I'm not going mad at least.

I've changed my drawing to give me more +- tolerance.

I guess instead of cutting in layers I could sequentially cut  using colour code order?

Over an landscape A3 what width column would you suggest?

regards

**Comment by Sean-anotherone** [comment]
It not so much then number of colours/layers, it's the number of fast jumps.   For your drawing I'd suggest trying 2x2 or 3x2 birds at each time, the original was 6x4 and with all the small parts probably had 100 jumps, so the cumulative error added up.

Yahoo Mail: Search, organise, conquer 
 
  On Sat, 23 Mar 2024 at 12:45, ***@***.***> wrote:   


Thanks for that. I'm glad I'm not going mad at least.

I've changed my drawing to give me more +- tolerance.

I guess instead of cutting in layers I could sequentially cut using colour code order?

Over an landscape A3 what width column would you suggest?

regards

**Comment by DaveBenji** [comment]
Wow! that's going to take me ages to cut any quantity. Let's hope we can fix it soon. :(

**Comment by DaveBenji** [comment]
okay. I've amended my drawing to cut in groups of six. Strangely enough according to the preview its quicker! I'll test run it on Wednesday and let you know  how I get on. regards

**Comment by magman2112** [comment]
I sense a little bit of confusion here perhaps.

By layers, I mean the colour you choose from the strip of them at the bottom of the screen, Lightburn calls these layers. You can do one laser operation per layer. There is no significant time issue in using multiple layers during cutting, apart from your time setting up your file in Lightburn. In engraving (fill) this can be quite a bit different.

I would still be tempted to group your drawings in columns to get the best results, as there will likely be a small amount of cumulative error for the movement between two columns of birds if you group them in sixes instead of threes (using your SVG file you provided above as a reference). The order that the layers will be cut is also significant, so if you cut all of the layers for the part between the legs in order after the body cuts, then you haven’t really changed everything.

Here’s a mock up of a cutting order which I think would give the best results.

Layer    Part
  1 —— Column 1 hole and between legs part
  2 —— Column 1 Main body cuts
  3 —— Column 2 hole and between legs part
  4 —— Column 2 Main body cuts
  5 —— Column 3 hole and between legs part
  6 —— Column 3 Main body cuts
Etc.

I appreciate that this is quite difficult to explain in this medium, but I hope this makes things a little clearer.

**Comment by DaveBenji** [comment]
Yes that's exactly what I have done. Ill let you know how I get on. David

**Comment by JackiePease** [comment]
I tried to use Gerald today but he still seems to be having dimensioning issues. I cut out a circle which should have been 26mm and it came out as 22mm. 

The x and y dimensions don't match either (I'd say it's out by about 3mm in 100mm)

**Comment by magman2112** [comment]
I looked at this on Tuesday night and found a discrepancy on the X-axis again. A 400mm test cut measured as 390mm, which is a 5% error. I also checked the Y-axis and this was measured as less than 0.5mm error in 400mm, so better than 99.5% accuracy.

I suspect this is a hangover from the changes we made last Thursday night. 

I adjusted the scaling in the firmware and Gerald is now cutting to correct dimensions again.

I would suggest that anyone using Gerald should do some test cuts before committing to cut large sheets of material with their artwork in the near future, until we find a solution to the current creep issue on the X-axis

**Comment by DaveBenji** [comment]
Hello. I worked on Gerald yesterday. I broke the work down into smaller sections using multiple Lightburn layers (line colours) into local cut groups as you suggested and it was better but there was still XY creep but, with a bit of scalpel assistant I was able to use the large majority of what I cut.

**Comment by DaveBenji** [comment]
Frustratingly inaccurate. The discrepancy of length on the x axis is back with a minus measurement from the drawing. Even with grouping of work in Lightburn colour layers the end point and start point are misaligned.

**Comment by magman2112** [comment]
We have now fitted a new Driver for the X-Axis stepper motor and in our initial tests we found that the creep no longer present. We will do some further tests on Maker Night, including re-running @DaveBenji example files again, to see if there is any improvement with this data.

**Comment by magman2112** [comment]
Closing all issues related to Gerald as this machine is now out of commission and will hopefully find a new home soon.
